src/biz/workspace/page_view.rs

```diff
@@ -32,15 +35,170 @@ use crate::biz::collab::{
 
 use super::ops::{broadcast_update, collab_from_doc_state};
 
+struct FolderUpdate {
+  pub updated_encoded_collab: Vec<u8>,
+  pub encoded_updates: Vec<u8>,
+}
+
+pub async fn create_page(
+  pg_pool: &PgPool,
+  collab_storage: &CollabAccessControlStorage,
+  uid: i64,
+  workspace_id: Uuid,
+  parent_view_id: &str,
+  view_layout: &ViewLayout,
+) -> Result<Page, AppResponseError> {
```

Comment:
as much as possible, we should use `AppResponseError` only at the `api` crate, business logic in `biz` should use `AppError` instead